We have obtained an optimum Leiden cluster (PCC threshold = 0.95, resolution limit = 1.1). This cluster yielded 4 communities, 
and the patient IDs and micro_diagnosis of each member in these communities has been stored in a .csv file. Using this information,
we will now perform **Deep Dive Analysis**. No Deep Dive Analysis will be performed on the optimum K-means cluster simply
because the obtained Kappa concordance statistic was too low (roughly 0.3).

**Sensitivity and Specificity**

Among the 4 communities produced, Community 1 yielded the largest number of Viral patients. So, we assume that the Leiden
Clustering is a diagnostic test, with the categorisation of a patient into community 1 being a positive result, and the
categorisation of a patient into either communities 2, 3 or 4 being classed as a negative result. We subsequently
calculate the sensitivity and specificity of this diagnostic test (assuming that our micro_diagnosis labels are 
completely accurate). Here are our results:

- **Sensitivity: 0.702**
- **Specificity: 0.823**

**Limma**

We will now find the most Differentially Expressed Genes (DEGs) between these 4 clusters, using the Limma package.
The Limma package (Linear Model for Microarray Analysis) can only be given 2 groups of patients at a time. It compares
the gene expression data for both groups of patients, and identifies the Differentially Expressed Genes (given a specific
p-value and log Fold Change value).

Consequently, I have performed 4 contrasts:
- C1 vs C2+C3+C4 (Rest)
- C2 vs Rest
- C3 vs Rest
- C4 vs Rest

For each contrast, I have recorded the most upregulated DEGs and the most downregulated DEGs. The p-value used to 
determine whether a gene was differentially expressed was 0.05, after having been adjusted for multiple-testing
(via the FDR correction method). The logFC threshold for upregulated DEGs was +1, and for downregulated DEGs was -1.

I consequently ended up with a number of DEGs in each contrast, as seen in the bar plot below.

![image](https://github.com/user-attachments/assets/03e6c070-0e85-48bb-92b0-2e76929f579b)

After having obtained a list of upregulated and downregulated DEGs, I plugged these DEGs into the XGR package,
which outputted which pathways were being downregulated / upregulated. In instances where more than 10 pathways
were involved, only the Top 10 pathways (by Fold Enrichment) were shown in the dot plots.

Fold Enrichment is the value which tells you how overrepresented each pathway is among your DEG list for the Cluster 
(Compared to what you'd expect by chance). 

**Heatmaps**

For each contrast, all the upregulated and downregulated DEGs were combined together. Then, the top 50 DEGs
are extracted based on the change in absolute logFC values between Cluster ? Vs Rest. These DEGs are then
plotted on a heatmap, showing their level of expression.

To view all the geom_plots and the heatmaps, please run the code !

